src/System.Management.Automation/engine/parser/tokenizer.cs

```diff
@@ -1229,23 +1229,125 @@ private int SkipBlockComment(int i)
             return i;
         }
 
-        private static char Backtick(char c)
+        private string Backtick(char c)
         {
             switch (c)
             {
-                case '0': return '\0';
-                case 'a': return '\a';
-                case 'b': return '\b';
-                case 'e': return '\u001b';
-                case 'f': return '\f';
-                case 'n': return '\n';
-                case 'r': return '\r';
-                case 't': return '\t';
-                case 'v': return '\v';
-                default: return c;
+                case '0': return "\0";
+                case 'a': return "\a";
+                case 'b': return "\b";
+                case 'e': return "\u001b";
+                case 'f': return "\f";
+                case 'n': return "\n";
+                case 'r': return "\r";
+                case 't': return "\t";
+                case 'u': return ScanUnicodeEscapeSequence();
+                case 'v': return "\v";
+                default: return c.ToString();
             }
         }
 
+        private string ScanUnicodeEscapeSequence()
+        {
+            // Unicode replacement char, used to represent unknown, unrecognized or unrepresentable characters.
+            var unknownChar = ((char)0xffdd).ToString();
+            int escSeqStartIndex = _currentIndex - 2;
+            bool isBracketedSequence = false;
+            bool isTerminated = false;
+
+            char c = PeekChar();
+            if (c == '{')
+            {
+                SkipChar();
+                isBracketedSequence = true;
+            }
+
+            int maxNumberOfHexDigits = (isBracketedSequence ? 6 : 4);
+
+            // Scan hex chars after the Unicode escape sequence start.
+            var sb = GetStringBuilder();
+            for (int i = 0; i < maxNumberOfHexDigits; i++)
+            {
+                c = PeekChar();
```

Comment:
I think you can use `GetChar` here - you're expecting more than one character, then just call `UngetChar` when you went past what you consume.